cdap-proto/src/main/java/co/cask/cdap/proto/artifact/ArtifactUtil.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.artifact;
+
+import co.cask.cdap.api.artifact.ArtifactRange;
+import co.cask.cdap.api.artifact.ArtifactVersionRange;
+import co.cask.cdap.api.artifact.InvalidArtifactRangeException;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.id.NamespaceId;
+
+/**
+ * Utility class for parsing {@link ArtifactRange}
+ */
+public final class ArtifactUtil {
```

Comment:
for consistency with other utility classes, let's name this ArtifactRanges or something along those lines.